src/models/expenseModel.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-console */
+/* eslint-disable function-paren-newline */
+/* eslint-disable no-shadow */
+'use strict';
+
+const expenses = [];
+
+const expenseModel = {
+  createExpense(userId, spentAt, title, amount, category, note) {
+    const newExpense = {
+      userId,
+      id: expenses.length + 1,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    };
+
+    expenses.push(newExpense);
+
+    return newExpense;
+  },
+
+  getAllExpenses(filters = {}) {
+    const filteringTests = {
+      userId: (id, expense) => Number(id) === expense.userId,
+      id: (id, expense) => Number(id) === expense.id,
+      from: (to, expense) => new Date(expense.spentAt) > new Date(to),
+      to: (to, expense) => new Date(expense.spentAt) < new Date(to),
+      categories: (category, expense) => category === expense.category,
+    };
+
+    return expenses.filter((expense) =>
+      Object.entries(filters).reduce((result, [currKey, currVal]) => {
```

Comment:
Even better than using a .reduce here would be using an _every_ instead.
Why? _.reduce_ will iterate and test all query keys always, while an _.every_ will stop iterating once the test fails.
This could improve performance by reducing the amount of iterations/tests